web/src/main/java/cz/muni/fi/pa165/mvc/controllers/UserController.java

```diff
@@ -0,0 +1,163 @@
+package cz.muni.fi.pa165.mvc.controllers;
+
+
+import cz.muni.fi.pa165.dto.CreateRoleDTO;
+import cz.muni.fi.pa165.dto.CreateUserDTO;
+import cz.muni.fi.pa165.dto.RoleDTO;
+import cz.muni.fi.pa165.dto.UserDTO;
+import cz.muni.fi.pa165.facade.UserFacade;
+import cz.muni.fi.pa165.facade.RoleFacade;
+
+import cz.muni.fi.pa165.entity.Role;
+import cz.muni.fi.pa165.entity.Users;
+
+import cz.muni.fi.pa165.services.RoleService;
+import cz.muni.fi.pa165.services.UserService;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import javax.validation.Valid;
+import javax.inject.Inject;
+import java.time.LocalDate;
+
+import cz.muni.fi.pa165.EntityUtils;
+
+@Controller
+//@RequestMapping(value = "/user")
+@RequestMapping(value = {WebUrls.URL_USER})
+public class UserController {
+    @Inject
+    private RoleFacade roleFacade;
+    @Inject
+    private UserFacade userFacade;
+
+    @Inject
+    private UserService userService;
+    @Inject
+    private RoleService roleService;
+
+    /**
+     * list of users.
+     *
+     * @param model data to display
+     * @return JSP page name
+     */
+    @RequestMapping(value = "/", method = RequestMethod.GET)
+    public String list(Model model) {
+        model.addAttribute("user", userFacade.getAllUser());
+        return "user/list";
+    }
+
+    @RequestMapping(value = "/{id}/delete", method = RequestMethod.POST)
+    public String delete(@PathVariable long id, UriComponentsBuilder uriBuilder,
+                         RedirectAttributes redirectAttributes) {
+        UserDTO user = userFacade.getUserById(id);
+        try {
+            userFacade.removeUser(id);
+            redirectAttributes.addFlashAttribute("alert_success", "User \"" + user.getLastName() + "\" was deleted.");
+        } catch (Exception ex) {
+            redirectAttributes.addFlashAttribute("alert_danger",
+                    "User \"" + user.getLastName() + "\" cannot be deleted, it might be used for a user.");
```

Comment:
Ta chybová hláška nepatrně při mazání uživatelů nedává smysl - jasně, může být referencovaný u ticketu/bookingu, takže selhat to může jednoduše, ale asi by to chtělo přeformulovat.